sno/working_copy/postgis.py

```diff
@@ -471,6 +494,26 @@ def delete_features(self, dbcur, dataset, pk_iter):
 
         return feat_count
 
+    def drop_table(self, target_tree_or_commit, *datasets):
```

Comment:
I just realised that this is called by `reset()` without using a transaction. So the table drop gets committed before the table is recreated...

`reset()` should wrap all its changes in a `self.session()` to avoid the drop being exposed to other transactions until the table's been recreated/repopulated

Reply:
Basically _every_ sno command needs to be wrapped in a single transaction. Firstly so changes either fully apply or don't at all; second for a consistent view of the database (dataset tables are consistent between queries when concurrent updates are happening); third to make sure other connections has a consistent view of the database (ie: Sno's changes are all visible or none)

Reply:
Not done - wrapping the reset() in a single session breaks GPKG support :(. I'll fix this in a separate PR.